design-proposals/2023-02-08-code-organisation.md

```diff
@@ -0,0 +1,137 @@
+# MapLibre Native Code Organisation Design Proposal
+
+This proposal is to define code organisation that allows simple and efficient development and usage of MapLibre Native.
+
+## Motivation
+
+MapLibre Native is currently a monorepo that includes both the core functionality and the SDKs. It's not obvious what is used where andfirst-time contributors may also be repelled a bit by the complexity. This proposal is not necessarily about the splitting of repository (although it will be covered in a section) but in general about code organisation.
```

Comment:
```suggestion
MapLibre Native is currently a monorepo that includes both the core functionality and the SDKs. It's not obvious what is used where and first-time contributors may also be repelled a bit by the complexity. This proposal is not necessarily about the splitting of repository (although it will be covered in a section) but in general about code organisation.
```